davici.c

```diff
@@ -131,23 +104,21 @@ static void ip_and_port(char *ip_address, char *ip, char *port)
 	}
 }
 
-static int connect_in_and_fcntl(int fd, const char *path)
+static int connect_in_and_fcntl(int fd, const char *address)
 {
 	struct sockaddr_in addr;
 	int flags;
 
 	memset(&addr, 0, sizeof(addr));
 	addr.sin_family = AF_INET;
-	char *ip_address = malloc(strlen(path) * sizeof(char));
-	char *ip = malloc(strlen(path) * sizeof(char));
+	char *ip = malloc(strlen(address) * sizeof(char));
+	char *ip_address = malloc(strlen(address) * sizeof(char));
```

Comment:
just thinking do you need this? Can you not pass address directly to `ip_and_port`?

Reply:
I need to separate out the `ip_address` and `port` from the overall `address` as the `ip_and_port` function uses the `:` as a token separator to set the `ip_address` and `port`

Reply:
but you create `ip` and `ip_address` and then pass `ip_address` to `ip_and_port`, just wondered if you could pass `address` directly to `ip_and_port` but just seen it takes a `char*` not a `const char*`